modules/hisat2.cfmod.pl

```diff
@@ -0,0 +1,194 @@
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+# Get Options
+my $required_cores;
+my $required_mem;
+my $required_modules;
+my $help;
+my $result = GetOptions ("cores=i" => \$required_cores, "mem=s" => \$required_mem, "modules" => \$required_modules, "help" => \$help);
+
+# QSUB SETUP
+# --cores i = offered cores. Return number of required cores.
+if($required_cores){
+	print CF::Helpers::allocate_cores($required_cores, 7, 7);
+	exit;
+}
+# --mem. Return the required memory allocation.
+if($required_mem){
+	print CF::Helpers::allocate_memory($required_mem, '8G', '8G');
+	exit;
+}
+# --modules. Return csv names of any modules which should be loaded.
+if($required_modules){
+	print 'hisat2,samtools';
+	exit;
+}
+# --help. Print help.
+if($help){
+	print "".("-"x15)."\n Hisat2 Module\n".("-"x15)."\n
+Hisat2 is a spliced aligner for RNA-Seq data.  It is fast and memory
+efficient.
+This script works out the encoding of input files, guesses whether they're
+paired end or not and runs hisat2. Output is piped through samtools to
+generate BAM files.\n
+For further information, please run hisat2 --help\n\n";
+	exit;
+}
+
+# MODULE
+my $timestart = time;
+
+# Read in the input files from the run file
+my ($files, $runfile, $job_id, $prev_job_id, $cores, $mem, $parameters, $config_ref) = CF::Helpers::load_runfile_params(@ARGV);
+my %config = %$config_ref;
+
+# Check that we have a genome defined
+
+# TODO: Define a proper hisat2 path variable, even though it will probably
+# end up being the same as the bowtie2 one.
+
+if(!defined($config{bowtie2_path})){
+    if (defined($config{bowtie_path})){
+		warn "\n\n### CF Error: No bowtie2 path found in run file $runfile for job $job_id.\n Exiting.. ###\n";
+		
+		exit;
+	}  else  {
+
+	    warn "\n\n###CF Error: No bowtie2 path found in run file $runfile for job $job_id. Exiting.. ###";
+	    exit;
+	}
+} else {
+	warn "\nAligning against ".$config{bowtie2_path}."\n\n";
+}
+
+my $gtf = '';
+if(defined($config{gtf_path})){
+
+		my $splice_file = $config{gtf_path};
+		$splice_file =~ s/\.gtf$/_hisat2_splices.txt/;
```

Comment:
I'll remove this (bit hacky) and request a new reference type called `hisat2_splices`